### PR TITLE
Fix read exception for external hive table without resource name

### PR DESCRIPTION
**Pull request by miomiocat**
### Steps to reproduce:
1. create external hive table
```
CREATE EXTERNAL TABLE `t1` (
  `id` int,
  `name` string
) ENGINE=HIVE
PROPERTIES (
"database" = "starrocks_test",
"table" = "hdfs_orc_table",
"hive.metastore.uris" = "thrift://192.168.7.158:9083"
);

Query OK, 0 rows affected (0.01 sec)
```

2.  query this table, then it will throw a exception
```
select * from t1;

ERROR 1064 (HY000): get hive client failed, resource[null] not exists
```

### cause analysis：
1. both `metaCache` and `metaClient` are indexed by resource name in `HiveRepository`, but absent in this situation.
2. Inconsistenct and insufficient check about `table properties` while creating external hive table.
3. if hive external table is created with properties both `hive.metastore.uris` and `resource` specified, resource related info will overwrite `hive.metastore.uris` without any conflicts check or prompt.
4. if hive external table is created with properties `hive.metastore.uris` specified only, all of column related check like existence or  partition column filling will be skipped.  Then throw query exception like above example due to missing resource name.

### how to fix
1. Ignore the hive.metastore.uris properties.
2. The resource must be specified.

@imay @wyb

### CODE REVIEW

**Comment by gengjun-git** [comment]
@wyb Is there still a scenario where spark load creates a hive external table?

**Comment by gengjun-git** [comment]
I remember that the hive external table created in spark load only specifies hive.metastore.uris

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java [hunk 1]
```diff
@@ -329,11 +330,27 @@ private void validate(Map<String, String> properties) throws DdlException {
             throw new DdlException("property " + HIVE_METASTORE_URIS + " or " + HIVE_RESOURCE + " must be set");
         }
 
+        // 1. check conflicts if both hive.metastore.uris and resource are set.
+        // 2. try to set proper resource if hive.metastore.uris is specified and hive.resource not.
+        HiveResource hiveResource = null;
+        HiveResource hiveResourceByUris = null;
+        HiveResource hiveResourceByName = null;
         if (!Strings.isNullOrEmpty(hiveMetastoreUris)) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:338]: I think it is better to remove the interface to create table with hive.metastore.uris.
Users can only create tables by specifying resource.

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:338]: Just ignore the hive.metastore.uris properties. And the resource must be specified.

**Comment by wyb** [comment]
> @wyb Is there still a scenario where spark load creates a hive external table?

Yes, spark load can also use the external tables created by specifying resource, just need to put hive.metastore.uris into hiveProperties in HiveTable.

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java (hunk 1 above)
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:332]: done, give a prompt here if hive.metastore.uris is specified.

@gengjun-git @wyb PTAL

**Review by imay — DISMISSED** [review]

**Comment by gengjun-git** [comment]
Daily will report error after this PR is merged. We should change the create hive resource cases.

**Review by gengjun-git — DISMISSED** [review]

**Comment by imay** [comment]
> Daily will report error after this PR is merged. We should change the create hive resource cases.

@gengjun-git 
Could you change the test case and merge it after changing it?

**Comment by gengjun-git** [comment]
> > Daily will report error after this PR is merged. We should change the create hive resource cases.
> 
> @gengjun-git Could you change the test case and merge it after changing it?

OK

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java [hunk 2]
```diff
@@ -322,16 +322,17 @@ private void validate(Map<String, String> properties) throws DdlException {
...
         }
 
         if (!Strings.isNullOrEmpty(hiveMetastoreUris)) {
             copiedProps.remove(HIVE_METASTORE_URIS);
-            hiveProperties.put(HIVE_METASTORE_URIS, hiveMetastoreUris);
+            LOG.warn("property " + HIVE_METASTORE_URIS + " will be ignored " +
+                    "and hive table will be created by using property " + HIVE_RESOURCE + " only.");
         }
 
         if (!Strings.isNullOrEmpty(resourceName)) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:338]: Remove this check because it has already been guaranteed.

Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java]: done, PTAL

**Review by wyb — APPROVED** [review]

**Comment by gengjun-git** [comment]
Merge after this PR https://github.com/StarRocks/StarRocksTest/pull/30

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**